scripts/chandra/build_parent_sample.py

```diff
@@ -0,0 +1,126 @@
+            src_name = catalog['name'][(catalog['obsid'] == int(file.strip().split('/')[-1][0:27].strip().split('_')[0][-5:])) &
+               (catalog['obi'] == int(file.strip().split('/')[-1][0:27].strip().split('_')[1][0:3])) &
+               (catalog['region_id'] == int(file.strip().split('/')[-1][0:27].strip().split('_')[3][-4:]))]
+        else:
+            src_name = catalog['name'][(catalog['obsid'] == int(file.strip().split('/')[-1][0:24].strip().split('_')[0][-5:])) &
+               (catalog['obi'] == int(file.strip().split('/')[-1][0:24].strip().split('_')[1][0:3])) &
+               (catalog['region_id'] == int(file.strip().split('/')[-1][0:24].strip().split('_')[2][-4:]))]
+        output["name"].append(src_name.data[0])
+        output["obsid"].append(int(file.strip().split('/')[-1][0:24].strip().split('_')[0][-5:]))
+        output["obi"].append(int(file.strip().split('/')[-1][0:24].strip().split('_')[1][0:3]))
+        
+    # Return the results
+    return output
+
+
+def save_in_standard_format(catalog="catalog.hdf5", output_filename="parent_sample_xray.hdf5", chandra_data_path="./output_data/", delete_files=False):
+    """ Save the spectra in standard HDF5 format
+    """
+    os.makedirs(os.path.dirname(chandra_data_path), exist_ok=True)
+    
+    logger = logging.getLogger('sherpa')
+    logger.setLevel(logging.ERROR)
+    
+    # Process all files
+    spectra = Table(processing_fn(catalog, chandra_data_path))
+    catalog = Table(catalog)
+    
+    catalog['name'] = catalog['name'].astype(str)
+    catalog['obsid'] = catalog['obsid'].astype(int)
+    catalog['obi'] = catalog['obi'].astype(int)
+    spectra['name'] = spectra['name'].astype(str)
+    spectra['obsid'] = spectra['obsid'].astype(int)
+    spectra['obi'] = spectra['obi'].astype(int)
+    # Join on target id with the input catalog
+    catalog = join(catalog, spectra, keys=['name','obsid','obi'], join_type='inner')
+    
+    with h5py.File(chandra_data_path+output_filename, 'w') as hdf5_file:
+        for key in catalog.colnames:
+            # Check if the column data type is a string
+            if catalog[key].dtype.kind in ['U', 'S']:
+                # Encode Unicode string to byte string
+                encoded_strings = np.char.encode(catalog[key], 'utf-8')
+                hdf5_file.create_dataset(key, data=encoded_strings)
+            elif catalog[key].dtype == 'O':  # Column with object dtype, potentially variable-length arrays
+                dt = h5py.special_dtype(vlen=np.dtype('float64'))  # or float32 if more appropriate
+                dataset = hdf5_file.create_dataset(key, (len(catalog[key]),), dtype=dt)
+                for i, item in enumerate(catalog[key]):
+                    dataset[i] = item  # Assign the variable-length array directly
+            else:
+                # Directly save the column as a dataset for non-string types
+                hdf5_file.create_dataset(key, data=catalog[key])                
+    return
```

Comment:
```suggestion
    if delete_files:
        # Delete all files and folders in chandra_data_path except for the saved hdf5 file
        for root, dirs, files in os.walk(chandra_data_path):
            for file in files:
                if file != output_filename:
                    file_path = os.path.join(root, file)
                    os.remove(file_path)
            for dir in dirs:
                if dir != output_filename:
                    dir_path = os.path.join(root, dir)
                    # Pause for a second to avoid "Directory not empty" error
                    shutil.rmtree(dir_path)
    return
```